ls1intum/Artemis
# `Tutorial groups`: Allow tutors to track number of attending students per session

### Checklist
#### General
#### Server
#### Client

### Motivation and Context
The goal is to enable tutors to monitor attendance in each session, so we can later calculate and display the utilization of a tutorial group in a future release.

### Description
We added an editable input field to the tutorial group session table for the attendance information.

### Steps for Testing

Verify that instructors and assigned tutors can update attendance numbers and that these changes are reflected for students who cannot edit the numbers themselves. (Refer to the provided video and screenshot for reference).

### Review Progress
#### Code Review
#### Manual Tests

### Test Coverage
| Class/File | Branch | Line |
|------------|-------:|-----:|
| TutorialGroupSessionResoucre.java - updateAttendanceCount | 100% | 100% |
| tutorial-group-session-row.component.ts | 54% | 98% |
| tutorial-group-sessions-table.component.ts | 85% | 100% |



### Screenshots

https://user-images.githubusercontent.com/29718932/214103177-7c4ca71b-cab2-4e59-ae55-2985b49a7b0d.mp4

![image](https://user-images.githubusercontent.com/29718932/214103433-41616bba-0ade-442c-a927-6ba77af28a47.png)
